helper/predeployment/predeployment.go

```diff
@@ -0,0 +1,254 @@
+package predeployment
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/contracts/staking"
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+	"github.com/0xPolygon/polygon-edge/state"
+	itrie "github.com/0xPolygon/polygon-edge/state/immutable-trie"
+	"github.com/0xPolygon/polygon-edge/state/runtime"
+	"github.com/0xPolygon/polygon-edge/state/runtime/evm"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo/abi"
+	"io/ioutil"
+	"math"
+	"math/big"
+	"os"
+	"strings"
+)
+
+var (
+	errABINotFound              = errors.New("abi field not found in specified JSON")
+	errBytecodeNotFound         = errors.New("bytecode field not found in specified JSON")
+	errDeployedBytecodeNotFound = errors.New("deployed bytecode field not found in specified JSON")
+)
+
+type contractArtifact struct {
+	ABI              []byte // the ABI of the Smart Contract
+	Bytecode         []byte // the raw bytecode of the Smart Contract
+	DeployedBytecode []byte // the deployed bytecode of the Smart Contract
+}
+
+// generateContractArtifact generates contract artifacts based on the
+// passed in Smart Contract JSON ABI
+func generateContractArtifact(filepath string) (*contractArtifact, error) {
+	var (
+		artifact *contractArtifact
+		err      error
+	)
+
+	// Read from the ABI from the JSON file
+	jsonFile, err := os.Open(filepath)
+	if err != nil {
+		return artifact, err
```

Comment:
Why don't you return `nil, err`?

Reply:
Fixed:

[6d7f9ee](https://github.com/0xPolygon/polygon-edge/pull/536/commits/6d7f9eea0947cdbbd29c9d57256edd27f63dfbfc)